charts/vald/templates/common/deny-all.yaml

```diff
@@ -0,0 +1,28 @@
+#
+# Copyright (C) 2019-2023 vdaas.org vald team <vald@vdaas.org>
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+# Deny all the ingress/egress traffic
+{{- if .Values.defaults.networkPolicy.enabled }}
+kind: NetworkPolicy
+apiVersion: networking.k8s.io/v1
+metadata:
+  name: deny-all
+spec:
+  podSelector: {}
```

Comment:
It would be better to select only Vald components and namespaces.

Reply:
Is this something we really want to do?

Setting it up that way will allow all the components traffic to and from the outside by default, which raises security concerns if there is a bad configuration, etc. Instead, I think it makes more sense to deny all the traffic and then only allowing traffic for components that really need external traffics, such as `filter gateway`.

Reply:
Sorry, I misunderstood.

However, I think this method of denying all network In/Out and then opening the destination target only for specific components may affect other components that are not managed by Vald, such as components deployed in the same Namespace.

It would be more flexible to control the appropriate In/Out for each component.

Reply:
I get what you mean. Does this resolve your concern?
[5353240](https://github.com/vdaas/vald/pull/2022/commits/5353240aa0f777a0a0f0ca9f81d0222ca486918c)

Reply:
looks good, thank you.